staging/src/k8s.io/apiextensions-apiserver/third_party/forked/celopenapi/model/schemas.go

```diff
@@ -228,8 +229,12 @@ func WithTypeAndObjectMeta(s *schema.Structural) *schema.Structural {
 		s.Properties["metadata"].Properties["generateName"].Type == "string" {
 		return s
 	}
+	generic := s.Generic
+	if generic.Type != "object" {
+		generic.Type = "object"
```

Comment:
this will override an explicit type other than `object`, which seems wrong... should this be something like

```
if generic.Type == "" {
...
```

Reply:
Good catch.